packages/x-date-pickers-pro/src/internal/hooks/useMultiInputRangeField/useMultiInputDateRangeField.ts

```diff
@@ -84,36 +98,43 @@ export const useMultiInputDateRangeField = <TDate, TChildProps extends {}>({
     rangeValueManager.defaultErrorState,
   );
 
-  const startInputProps: UseDateFieldComponentProps<TDate, TChildProps> = {
+  const startFieldProps: UseDateFieldComponentProps<TDate, TTextFieldProps> = {
     error: !!validationError[0],
-    ...inStartTextFieldProps,
+    ...startTextFieldProps,
     disabled,
     readOnly,
     format,
+    unstableFieldRef: unstableStartFieldRef,
     value: valueProp === undefined ? undefined : valueProp[0],
     defaultValue: defaultValue === undefined ? undefined : defaultValue[0],
     onChange: handleStartDateChange,
+    selectedSections,
+    onSelectedSectionsChange,
   };
 
-  const endInputProps: UseDateFieldComponentProps<TDate, TChildProps> = {
+  const endFieldProps: UseDateFieldComponentProps<TDate, TTextFieldProps> = {
     error: !!validationError[1],
-    ...inEndTextFieldProps,
+    ...endTextFieldProps,
     format,
     disabled,
     readOnly,
+    unstableFieldRef: unstableEndFieldRef,
     value: valueProp === undefined ? undefined : valueProp[1],
     defaultValue: defaultValue === undefined ? undefined : defaultValue[1],
     onChange: handleEndDateChange,
+    selectedSections,
+    onSelectedSectionsChange,
   };
 
-  const startDateResponse = useDateField<TDate, TChildProps>({
-    props: startInputProps,
+  const startDateResponse = useDateField({
```

Comment:
![image](https://user-images.githubusercontent.com/3309670/221129943-6d333e50-f1cc-49b2-9a19-fb9b52f6eebd.png)

It does error out and I did not find how to avoid that one :/

Reply:
Gotcha. By the way, does `TTextFieldProps` not look a bit strange to you? 🙈 
Seems like both a generic and actual props type name at the same time... 🤯 😆

Reply:
How would you name it ?
I wanted to make clear that it's the prop passed to the `textField` and it does not contain the field-related props (validation props for example)

Reply:
Maybe one of these or something along these lines? 🤔 
- `ITextFieldProps` (`I` for interface—quite a popular approach on certain communities/languages)
- `TextFieldSlotProps` (Add something extra (`Slot`) to create more difference from the actual slot component props name)
- `TTextFieldSlotProps`